examples/carousel/js/carousel-tablist.js

```diff
@@ -0,0 +1,373 @@
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+
+  var index;
+
+  if (typeof currentTabNode !== 'object') {
+    currentTabNode = this.currentTabNode;
+  }
+
+  if (currentTabNode === this.firstTabNode) {
+    this.setSelectedTab(this.lastTabNode, moveFocus);
+  }
+  else {
+    index = this.tabNodes.indexOf(currentTabNode);
+    this.setSelectedTab(this.tabNodes[index - 1], moveFocus);
+  }
+}
+
+CarouselTablist.prototype.setSelectedToNextTab = function (currentTabNode, moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+
+  var index;
+
+  if (typeof currentTabNode !== 'object') {
+    currentTabNode = this.currentTabNode;
+  }
+
+  if (currentTabNode === this.lastTabNode) {
+    this.setSelectedTab(this.firstTabNode, moveFocus);
+  }
+  else {
+    index = this.tabNodes.indexOf(currentTabNode);
+    this.setSelectedTab(this.tabNodes[index + 1], moveFocus);
+  }
+}
+
+CarouselTablist.prototype.setSelectedTofirstTabNode = function (moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+  this.setSelectedTab(this.firstTabNode, moveFocus);
+}
+
+CarouselTablist.prototype.setSelectedTolastTabNode = function (moveFocus) {
+  if (typeof moveFocus != 'boolean') {
+    moveFocus = false;
+  }
+  this.setSelectedTab(this.lastTabNode, moveFocus);
+}
+
+CarouselTablist.prototype.rotateSlides = function () {
+  if (this.rotate) {
+    this.setSelectedToNextTab();
+  }
+  setTimeout(this.rotateSlides.bind(this), this.timeInterval);
+}
+
+CarouselTablist.prototype.updateRotation = function () {
+
+  // Pause rotation when keyboard focus enters the carousel
+  if (this.hasFocus) {
+    this.isStopped = true;
+  }
+
+  if (!this.hasHover && !this.hasFocus && !this.isStopped) {
+    this.rotate = true;
+    this.liveRegionNode.setAttribute('aria-live', 'off');
+  }
+  else {
+    this.rotate = false;
+    this.liveRegionNode.setAttribute('aria-live', 'polite');
+  }
+
+  if (this.isStopped) {
+    this.pauseButtonNode.setAttribute('aria-label', this.playLabel);
+    this.pauseButtonNode.classList.remove('pause');
+    this.pauseButtonNode.classList.add('play');
+  }
+  else {
+    this.pauseButtonNode.setAttribute('aria-label', this.pauseLabel);
+    this.pauseButtonNode.classList.remove('play');
+    this.pauseButtonNode.classList.add('pause');
+  }
+
+
+}
+
+CarouselTablist.prototype.toggleRotation = function () {
+  if (this.isStopped) {
+    if (!this.hasHover && !this.hasFocus) {
+      this.isStopped = false;
+    }
+  }
+  else {
+    this.isStopped = true;
+  }
+
+  this.updateRotation();
+
+}
+
+  /* Event Handlers */
+
+CarouselTablist.prototype.handleImageLinkFocus = function () {
+  this.liveRegionNode.classList.add('focus');
+}
+
+CarouselTablist.prototype.handleImageLinkBlur = function () {
+  this.liveRegionNode.classList.remove('focus');
+}
+
+CarouselTablist.prototype.handleMouseOver = function (event) {
+  if (!this.pauseButtonNode.contains(event.target)) {
```

Comment:
It seems a bit odd to me that the carousel is paused on hover, but then has an exception for when the cursor is directly over the pause button.

I actually thought it was a bug, particularly when the carousel was playing and then unexpectedly stopped after I'd accidentally knocked the mouse a few pixels over without noticing.